python-lib/type_conversion.py

```diff
@@ -0,0 +1,220 @@
+        self.mapping_dss_to_hyper = {
+            'array': (SqlType.text(), handle_nan(str)),
+            'bigint': (SqlType.int(), handle_nan(int)),
+            'boolean': (SqlType.bool(), handle_nan(bool)),
+            'date': (SqlType.timestamp(), handle_nat(to_hyper_timestamp)),
+            'double': (SqlType.double(), handle_nan(float)),
+            'float': (SqlType.double(), handle_nan(float)),
+            'geometry': (SqlType.text(), handle_nan(str)),
+            'geopoint': (SqlType.geography(), handle_nan(to_hyper_geography)),
+            'int': (SqlType.int(), handle_nan(int)),
+            'map': (SqlType.text(), handle_nan(str)),
+            'object': (SqlType.text(), handle_nan(str)),
+            'smallint': (SqlType.int(), handle_nan(int)),
+            'string': (SqlType.text(), handle_nan(str)),
+            'tinyint': (SqlType.int(), handle_nan(int)),
+        }
+
+        # Mapping Tableau Hyper to DSS types
+        self.mapping_hyper_to_dss = {
+            TypeTag.BIG_INT: ('bigint', lambda x: None if math.isnan(x) else int(x)),
+            TypeTag.BYTES: ('string', lambda x: None if x is None else str(x)),
+            TypeTag.BOOL: ('bool', lambda x: None if x is None else bool(x)),
+            TypeTag.CHAR: ('string', lambda x: None if x is None else str(x)),
+            TypeTag.DATE: ('date', lambda x: None if x is None else to_dss_date(x)),
+            TypeTag.DOUBLE: ('double', lambda x: None if math.isnan(x) else float(x)),
+            TypeTag.GEOGRAPHY: ('geopoint', lambda x: None if x is None else to_dss_geopoint(x)),
+            TypeTag.INT: ('int', lambda x: None if math.isnan(x) else int(x)),
+            TypeTag.INTERVAL: ('string', lambda x: None if x is None else str(x)),
+            TypeTag.JSON: ('string', lambda x: None if x is None else str(x)),
+            TypeTag.NUMERIC: ('double', lambda x: None if math.isnan(x) else float(x)),
+            TypeTag.OID: ('string', lambda x: None if x is None else str(x)),
+            TypeTag.SMALL_INT: ('int', lambda x: None if math.isnan(x) else int(x)),
+            TypeTag.TEXT: ('string', lambda x: None if x is None else str(x)),
+            TypeTag.TIME: ('string', lambda x: None if x is None else str(x)),
+            TypeTag.TIMESTAMP: ('date', lambda x: None if x is None else to_dss_timestamp(x)),
+            TypeTag.TIMESTAMP_TZ: ('string', lambda x: None if x is None else str(x)),
+            TypeTag.VARCHAR: ('string', lambda x: None if x is None else str(x))
+        }
+
+    def dss_type_to_hyper(self, dss_type):
+        """
+        Convert an identifier (string) of a single dss storage type to the mapped hyper type.
+
+        :param dss_type: Storage type string identifier from DSS
+            examples:
+            >>> "bigint"
+        :return: The mapped Tableau Hyper Type
+            examples:
+            >>> SqlType.big_int()
+        """
+        if dss_type not in self.mapping_dss_to_hyper:
```

Comment:
Suboptimal, there is 3 lookup for the same key.
better to do:
```
try:
  toto = self.mapping_dss_to_hyper[dss_type]
  if toto is None:
     //pas content
  else:
     ...
except:
  //error blabla
```

Reply:
Changed to 
```
        try:
            (dss_type, function) = self.mapping_hyper_to_dss.get(hyper_type, (None, None))
            if dss_type is None:
                logger.warning("Invalid DSS storage type {}".format(hyper_type))
                raise ValueError("Invalid DSS storage type {}".format(hyper_type))
            else:
                return dss_type
        except Exception as err:
            raise err
```